src/coreclr/jit/emitarm64.cpp

```diff
@@ -1396,8 +1396,40 @@ void emitter::emitInsSanityCheck(instrDesc* id)
         case IF_SVE_EA_1A: // ........xx...... ...iiiiiiiiddddd -- SVE broadcast floating-point immediate (unpredicated)
             assert(insOptsScalableAtLeastHalf(id->idInsOpt()));
             assert(isVectorRegister(id->idReg1()));                           // ddddd
+            assert(isValidUimm8(emitGetInsSC(id)));                           // iiiiiiii
+            assert(isValidVectorElemsize(optGetSveElemsize(id->idInsOpt()))); // xx
+            break;
+
+        case IF_SVE_EB_1A: // ........xx...... ..hiiiiiiiiddddd -- SVE broadcast integer immediate (unpredicated)
+            assert(insOptsScalableStandard(id->idInsOpt()));
+            // Size specifier must be able to fit left-shifted immediate
+            assert(insOptsScalableAtLeastHalf(id->idInsOpt()) || !id->idOptionalShift());
```

Comment:
not sure if I follow why we need `insOptsScalableAtLeastHalf(id->idInsOpt()` here? `EB_1A` is either https://docsmirror.github.io/A64/2023-06/dup_z_i.html or https://docsmirror.github.io/A64/2023-06/mov_dup_z_i.html and they both take `B`

Reply:
You're correct that `B` is accepted. I believe if the immediate value is being left-shifted, the size specifier has to be at least 16 bits. The docs you linked say this: "The immediate operand is a signed value in the range -128 to +127, and for element widths of 16 bits or higher it may also be a signed multiple of 256 in the range -32768 to +32512 (excluding 0)."

cstool's behavior doesn't seem to match the documentation exactly, though. From my local testing, if the immediate is left-shifted but the specifier is `B`, cstool refuses to parse the instruction, even if the immediate being shifted is 0. I wrote this assert to match the behavior of cstool, such that if `id->idOptionalShift` is true, the size specifier cannot be `B`. @a74nh does this sound correct to you?

Reply:
IMO, we should go with the arm specs. did you try validating it with LATE_DISASM?

Reply:
got it. Let's wait for @a74nh to comment on this.

Reply:
>You're correct that B is accepted. I believe if the immediate value is being left-shifted, the size specifier has to be at least 16 bits. The docs you linked say this: "The immediate operand is a signed value in the range -128 to +127, and for element widths of 16 bits or higher it may also be a signed multiple of 256 in the range -32768 to +32512 (excluding 0)."

Yes.  The optional shift is being used to specify the range -32768 to +32512. That range is only valid for widths of `H` or wider. Therefore, it's not valid to specify a shift with size `B`.

It also makes sense as a value shifted by 8 cannot fit into a `B`, and so would always be 0.

> if the immediate is left-shifted but the specifier is B, cstool refuses to parse the instruction, even if the immediate being shifted is 0.

That makes sense. It's still trying to left shift 0 by 8, and the documentation says left shift isn't valid for `B`. At a microarchitecture level, it wouldn't to useful encode a special case just to allow that.

I'm happy with the asserts.